ivy-data-testing/src/test/java/com/ivy/data/testing/FakeAccountRepositoryTest.kt

```diff
@@ -0,0 +1,355 @@
+package com.ivy.data.testing
+
+import com.ivy.data.model.Account
+import com.ivy.data.model.AccountId
+import com.ivy.data.model.primitive.AssetCode
+import com.ivy.data.model.primitive.ColorInt
+import com.ivy.data.model.primitive.NotBlankTrimmedString
+import io.kotest.core.spec.style.FreeSpec
+import io.kotest.matchers.shouldBe
+import java.time.Instant
+import java.util.UUID
+
+class FakeAccountRepositoryTest : FreeSpec({
+    fun newRepository() = FakeAccountRepository()
+
+    "find by id" - {
+        "existing id" {
+            // given
+            val repository = newRepository()
+            val id = AccountId(UUID.randomUUID())
+            val account = Account(
+                id = id,
+                name = NotBlankTrimmedString("Bank"),
+                asset = AssetCode("BGN"),
+                color = ColorInt(1),
+                icon = null,
+                includeInBalance = true,
+                orderNum = 1.0,
+                lastUpdated = Instant.EPOCH,
+                removed = false
+            )
+
+            // when
+            repository.save(account)
+            val res = repository.findById(id)
+
+            // then
+            res shouldBe account
+        }
+
+        "not existing id" {
+            // given
+            val repository = newRepository()
+            val id = AccountId(UUID.randomUUID())
+
+            // when
+            val res = repository.findById(id)
+
+            // then
+            res shouldBe null
+        }
+    }
+
+    "find all" - {
+        "not deleted accounts" {
+            // given
+            val repository = newRepository()
+            val id1 = AccountId(UUID.randomUUID())
+            val id2 = AccountId(UUID.randomUUID())
+            val account1 = Account(
+                id = id1,
+                name = NotBlankTrimmedString("Bank"),
+                asset = AssetCode("BGN"),
+                color = ColorInt(1),
+                icon = null,
+                includeInBalance = true,
+                orderNum = 1.0,
+                lastUpdated = Instant.EPOCH,
+                removed = false
+            )
+            val accounts = listOf(
+                account1,
+                Account(
+                    id = id2,
+                    name = NotBlankTrimmedString("Cash"),
+                    asset = AssetCode("BGN"),
+                    color = ColorInt(1),
+                    icon = null,
+                    includeInBalance = true,
+                    orderNum = 2.0,
+                    lastUpdated = Instant.EPOCH,
+                    removed = true
+                )
+            )
+
+            // when
+            repository.saveMany(accounts)
```

Comment:
This also needs to test that `findAll` sorts accounts by id. Put `.sortedByDescending()` to intentionally mess up the accounts order.